plugins/micromega/zify.ml

```diff
@@ -1005,8 +1017,19 @@ type typed_constr = {constr : EConstr.t; typ : EConstr.t; inj : EInjT.t}
 
 let get_injection env evd t =
   try
-    match snd (ConstrMap.find evd t !table_cache) with
-    | InjTyp i -> i
+    let ((_,i),b) =
+      match ConstrMap.find_all evd t !table_cache with
+      | []  -> assert false
+      | [e] -> (e,false)
+      | e::_ -> (e,true) in
+    match i with
+    | InjTyp i ->
+       if b then
+         Feedback.msg_warning
+           Pp.(str "Multiple injections for "++
+                 pr_constr env evd t ++ str ". Picking " ++
+                 pr_constr env evd i.decl ++ str".");
```

Comment:
Can we give this warning a category and make it silencable?

Reply:
> Can we give this warning a category and make it silencable (and also able to be turned into an error)?

I noticed that this is actually dead code. I have removed it completely.